src/cdk-experimental/listbox/listbox.ts

```diff
@@ -6,19 +6,79 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {Directive} from '@angular/core';
+import {
+  ContentChildren,
+  Directive,
+  ElementRef, EventEmitter, forwardRef, Inject,
+  Input, Output,
+  QueryList
+} from '@angular/core';
 
+/**
+ * Directive that applies interaction patterns to an element following the aria role of option.
+ * Typically meant to be placed inside a listbox. Logic handling selection, disabled state, and
+ * value is built in.
+ */
 @Directive({
   selector: '[cdkOption]',
   exportAs: 'cdkOption',
   host: {
     role: 'option',
+    '(click)': '_onClickUpdateSelected()',
+    '[attr.aria-selected]': '_selected || null',
+    '[attr.data-optionid]': '_optionId',
   }
 })
 export class CdkOption {
+  private _selected: boolean = false;
+  private _optionId: string;
+
+  /** Whether the option is selected or not */
+  @Input()
+  get selected(): boolean {
+    return this._selected;
+  }
+  set selected(value: boolean) {
+    this._selected = value;
+  }
+
+  constructor(private readonly _elementRef: ElementRef,
+              @Inject(forwardRef(() => CdkListbox)) public listbox: CdkListbox) {
+    this.setOptionId(`cdk-option-${_uniqueIdCounter++}`);
+  }
+
+  _onClickUpdateSelected() {
```

Comment:
I would combine this method and `toggleSelected` into a single method just called `toggle`
```ts
toggle() {
   this.selected = !this.selected;
  this.listbox._emitChangeEvent(this);
}
```

Reply:
So for a function as simple as this one, I suppose its not necessary to have a function name that is more descriptive than just "toggle"

Reply:
Yeah, my philosophy is to name methods for what they do rather than how they're called